src/diff/children.js

```diff
@@ -132,7 +121,8 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
  */
 export function toChildArray(children, flattened, map) {
 	if (flattened == null) flattened = [];
-	if (children==null || typeof children === 'boolean') {}
+	if (children===undefined) {}
```

Comment:
Hmm why special case `undefined` here? Would it work for something like the following? While probably less common, I think it is still valid.

```tsx
function Parent(props) {
	let sibling;
	if (props.condition) {
		sibling = <Component />;
	}

	return (
		<div>
			<div>Hello</div>
			{ sibling }
		</div>
	);
}
```

Reply:
That's a good point. I did this because otherwise components without any children at all will always end up with `[null]` as children.

```jsx
const App = () => "Hello World";
render(<App />, dom); // vnode._children will be [null]
```

Need to think about this a bit more. The pattern you suggested makes sense, but we need a way to not do more work than necessary when there are no children at all.

Reply:
Just checked with react and they coerce `undefined` to `null` too.